main.cpp

```diff
@@ -209,17 +292,77 @@ struct ElevatorSystem
     ElevatorCar car3;        // embedded structure
     ElevatorCar car4;        // embedded structure    
 
-
     // move down from current floor 
-    void floorDown( int downTarget );
+    // void floorDown( int downTarget );
+    void e1FloorDown ( ElevatorCar car1 );
+    void e2FloorDown ( ElevatorCar car2 );
+    void e3FloorDown ( ElevatorCar car3 );
+    void e4FloorDown ( ElevatorCar car4 );
     // move up from current floor
-    void floorUp( int upTarget ); 
-
+    // void floorUp( int upTarget ); 
+    void e1FloorUp ( ElevatorCar car1 );
+    void e2FloorUp ( ElevatorCar car2 );
+    void e3FloorUp ( ElevatorCar car3 );
+    void e4FloorUp ( ElevatorCar car4 );
```

Comment:
Are you aware that the `carN` being passed to these functions is different from the member variable with the same name? 

Your implementations make me think you want to modify the member variables of `ElevatorSystem`, but instead you're modifying the instance being passed to the function.   send me a DM and let me know your thoughts on this set of functions you wrote.